### PR TITLE
Analytics SetDefaultParameters

### PR DESCRIPTION
**Pull request by jonsimantov**
Add SetDefaultParameters API to Firebase Analytics Unity SDK.

This commit introduces the `FirebaseAnalytics.SetDefaultParameters()` method to the Unity SDK, wrapping the underlying C++ `firebase::analytics::SetDefaultEventParameters()` and `firebase::analytics::ClearDefaultEventParameters()` functions.

***

Calls to the new API have been integrated into the existing TestApp (UIHandler.cs) for manual verification and TestAppAutomated (UIHandlerAutomated.cs) for automated execution. These tests will exercise the API surface but not programmatically verify results from the Unity side.

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [X] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: analytics/src/FirebaseAnalytics.cs [hunk 1]
```diff
@@ -15,11 +15,14 @@
  */
 
 using System.Threading.Tasks;
+using Firebase.Platform;
 
 namespace Firebase.Analytics {
 
 public static partial class FirebaseAnalytics {
 
+  private static readonly Firebase.Platform.ModuleLogger logger = new Firebase.Platform.ModuleLogger(typeof(FirebaseAnalytics).ToString());
```
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs]: Let's hard code the tag instead of using typeof().ToString.

File: analytics/src/FirebaseAnalytics.cs [hunk 2]
```diff
@@ -248,6 +251,59 @@ public static void SetConsent(System.Collections.Generic.IDictionary< ConsentTyp
     FirebaseAnalyticsInternal.SetConsentWithInts(consentSettingsMap);
   }
 
+  /// @brief Sets default event parameters.
+  ///
+  /// These parameters are logged with all events, in addition to the parameters passed to the
+  /// LogEvent() call. They are useful for logging common parameters with all events.
+  /// Default event parameters are overridden by event-specific parameters if the names are the
+  /// same. Default parameters are removed if the dictionary is null or empty.
```
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs:259]: Let's change it so that default parameters are only cleared if the dictionary is null. *Individual* parameters will be removed (by the underlying implementation) if the value of a given key is null.

File: analytics/src/FirebaseAnalytics.cs [hunk 3]
```diff
@@ -248,6 +251,59 @@ public static void SetConsent(System.Collections.Generic.IDictionary< ConsentTyp
...
+  ///
+  /// @param[in] parameters The dictionary of parameters to set.
+  ///   If null, clears all default parameters.
+  public static void SetDefaultEventParameters(
+      System.Collections.Generic.IDictionary<string, object> parameters) {
+    if (parameters == null || parameters.Count == 0) {
```
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs]: Don't do this if count = 0, only if it's null.

File: analytics/src/FirebaseAnalytics.cs [hunk 4]
```diff
@@ -248,6 +251,59 @@ public static void SetConsent(System.Collections.Generic.IDictionary< ConsentTyp
...
+              kvp.Key, e.ToString()));
+          // If adding to parameterNames succeeded but Variant.FromObject failed,
+          // we need to remove the key that was added optimistically.
+          if (parameterNames.Count > parameterValues.Count) {
+              parameterNames.RemoveAt(parameterNames.Count - 1);
+          }
+        }
+      }
+
+      if (parameterNames.Count == 0 && originalCount > 0) {
```
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs]: I don't think we need this originalCount logic any more if we change it so an empty dictionary is just passed through verbatim.

File: analytics/src/FirebaseAnalytics.cs [hunk 5]
```diff
@@ -248,6 +251,59 @@ public static void SetConsent(System.Collections.Generic.IDictionary< ConsentTyp
...
+      logger.LogMessage(LogLevel.Debug, "Cleared default event parameters.");
+    } else {
+      StringList parameterNames = new StringList();
+      VariantList parameterValues = new VariantList();
+      int originalCount = parameters.Count; // Store original count for later check
+
+      foreach (var kvp in parameters) {
+        try {
+          parameterNames.Add(kvp.Key);
+          parameterValues.Add(Firebase.Variant.FromObject(kvp.Value));
```
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs]: Wouldn't it be cleaner to do the conversion first, and then add the name and value only once the conversion succeeds? This will prevent us from having to remove the value in the exception handler below.

File: analytics/src/swig/analytics.i [hunk 6]
```diff
@@ -100,6 +118,9 @@ void SetConsentWithInts(const std::map<int, int>& settings) {
 %ignore firebase::analytics::LogEvent(const char*, const Parameter*, size_t);
 // Ignore SetConsent, in order to convert the types with our own function.
 %ignore firebase::analytics::SetConsent;
+// Ignore SetDefaultEventParameters and ClearDefaultEventParameters, as we handle them
```
Comment (jonsimantov) [inline analytics/src/swig/analytics.i]: This comment is outdated as we don't ignore ClearDefaultEventParameters.

File: analytics/testapp/Assets/Firebase/Sample/Analytics/UIHandlerAutomated.cs [hunk 7]
```diff
@@ -108,6 +109,34 @@ Task TestAnalyticsSetConsentDoesNotThrow() {
...
+      FirebaseAnalytics.SetDefaultEventParameters(new Dictionary<string, object>
+      {
+          { "auto_default_param_string", "auto_value_1" }
+      });
+
+      DebugLog("Automated Test: Setting multiple default parameters: {'auto_default_param_int', 789, 'auto_default_param_double', 12.34, 'auto_default_param_bool', false}");
+      FirebaseAnalytics.SetDefaultEventParameters(new Dictionary<string, object>
+      {
+          { "auto_default_param_int", 789 },
+          { "auto_default_param_double", 12.34 },
```
Comment (jonsimantov) [inline analytics/testapp/Assets/Firebase/Sample/Analytics/UIHandlerAutomated.cs:126]: We should also test with a value of null to make sure that that's passed through correctly.

File: analytics/src/FirebaseAnalytics.cs (hunk 5 above)
Comment (jonsimantov) [inline analytics/src/FirebaseAnalytics.cs]: Also can we just confirm that Firebase.Variant.FromObject supports null properly, and will output a wrapped Variant::Null()?
